src/optimizer/index.js

```diff
@@ -30,13 +32,14 @@ module.exports = {
       ? transformsWhitelist
       : Object.keys(optimizationTransforms);
 
+    let result = new transform.TransformResult(parser.parse(regexp));
```

Comment:
The `optimize` potentially can accept the AST directly, which won't work with the `parse.parse`; we should consider this use-case here.